### PR TITLE
Introduces a new download endpoint, adds upload support to import

### LINKED ISSUE #1611: Update endpoints and frontend to upload/download files outside of electron
## Abstract

I did a quick check for the backend endpoints, and unless I am missing something there are only two endpoints that the frontend passes paths directly:

- `/api/1/history/export/` writes to the filesystem
- `/api/1/import` reads from the file system

We need to allow these endpoints to operate as upload when reading and download when writing.

#### Export
Export can take a optional parameter e.g. `/api/1/history/export/?download=true` and return the csv file in the response body when the flag is set.

Proper headers have to be set. https://stackoverflow.com/questions/393647/response-content-type-as-csv

#### Import
Import needs to also support uploading. The current import API uses the `PUT` verb so we could make a second call with the `POST` verb that accepts a body with the file that need to be imported.

### Tasks
- Update backend
- Update frontend

### PR DESCRIPTION
**Pull request by kelsos**
Closes #1611 

Testing the import:
`curl -v -X POST -H "Content-Type: multipart/form-data" -F "source=crypto.com" -F "file=@cryptocom_trades_list.csv" http://localhost:4242/api/1/import`

for the download you can open it in the browser after generating the report going to `http://localhost:4242/api/1/history/download` in the browser should start downloading a report.zip. The zip should contain any csv files generated

### CODE REVIEW

File: rotkehlchen/csv_exporter.py [hunk 1]
```diff
@@ -508,3 +512,32 @@ def create_files(self, dirpath: Path) -> Tuple[bool, str]:
             return False, str(e)
 
         return True, ''
+
+    def create_zip(self) -> Optional[str]:
+        if not self.create_csv:
+            return None
+        dirpath = Path(mkdtemp())
```
Comment (LefterisJP) [inline rotkehlchen/csv_exporter.py:519]: Add a comment and issue about managing to delete this once it's downloaded. Normally (as also seen in other places in the code) we would have used a context manager (a with statement) with this: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

File: rotkehlchen/api/rest.py [hunk 2]
```diff
@@ -1984,3 +2003,23 @@ def sync_data(
...
+    def import_data_from_file(
+            self,
+            source: Literal['cointracking.info', 'crypto.com'],
+            file: FileStorage,
+    ) -> Response:
+        if file.filename is None or not file.filename.endswith('.csv'):
+            result = wrap_in_fail_result(f'{file.filename} is not a csv')
+            return api_response(result, status_code=HTTPStatus.BAD_REQUEST)
+
+        temp_directory = Path(mkdtemp())
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:2017]: Don't use mkdtemp here. Use https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

We already use it in coinbasepro if you wanna look at an example: https://github.com/rotki/rotki/blob/31cd9b0d5cbe56777a36adafc817fba54abd3d58/rotkehlchen/exchanges/coinbasepro.py#L558

But the python docs are also good.

If you do that the cleanup at the end will not be needed at all.

**Review by LefterisJP — COMMENTED** [review]
First comments from the code only

File: rotkehlchen/api/rest.py [hunk 3]
```diff
@@ -1116,6 +1119,22 @@ def export_processed_history_csv(self, directory_path: Path) -> Response:
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
+    @require_loggedin_user()
+    def download_processed_history_csv(self) -> Response:
+        if len(self.rotkehlchen.accountant.csvexporter.all_events_csv) == 0:
+            result_dict = wrap_in_fail_result('No history processed in order to perform an export')
+            return api_response(result_dict, status_code=HTTPStatus.CONFLICT)
```
[the remark this answers is not in the document]
Comment (kelsos) [reply rotkehlchen/api/rest.py:1126]: Let's keep it like this to differentiate.

File: rotkehlchen/csv_exporter.py (hunk 1 above)
Comment (kelsos) [reply rotkehlchen/csv_exporter.py:520]: Issue created #1853

**Review by LefterisJP — COMMENTED** [review]
Another thing that needs handling is a better error when the tax report has generated nothing and the user press to download csv. I got take to this when I pressed it:

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**